### ISSUE TITLE
Error prop on error page misses URL field in Nuxt 3.4.3

### ISSUE THREAD

**Issue by Citingdude** [nuxt/nuxt#20762]
### Environment

- Operating System: Linux
- Node Version:     v16.14.2
- Nuxt Version:     3.4.3
- Nitro Version:    2.4.0
- Package Manager:  npm@7.17.0
- Builder:          vite
- User Config:      -
- Runtime Modules:  -
- Build Modules:    -

### Reproduction

https://stackblitz.com/edit/github-qoy5uj?file=middleware/locale.global.ts

You have to trigger the error page to see the result. Go to "/test" for example.

### Describe the bug

In the new Nuxt 3.4.3 version, the error prop on the error page misses the URL field. This field used to be in 3.4.2 and we used it to handle redirections. 

This only happens when we use useCookie composable in a global middleware (see reproduction).

Before:

```json 
{
  "error": {
    "url": "/resrse",
    "statusCode": 404,
    "statusMessage": "Page not found: /resrse",
    "message": "Page not found: /resrse",
    "stack": ""
  }
}
```

After: 
``` json
{
  "error": {
    "message": "Page not found: /fdfds",
    "statusCode": 404,
    "statusMessage": "Page not found: /fdfds"
  }
}
```

**Comment by darioferderber** [comment]
@Citingdude I think this is fixed with 16bf228437736099f07e678a1c0628d829f56c47.
Hopefully, it should work with the new release.

**Comment by Lehoczky** [comment]
I've updated Nuxt in your reproduction and the issue seems to be fixed. Can you please confirm that in `^3.5.0` everything works for you as well?

![image](https://github.com/nuxt/nuxt/assets/31937175/baf2dc48-548c-4b24-ad76-8283de626024)

![image](https://github.com/nuxt/nuxt/assets/31937175/fc561c43-b8ce-401b-993d-b08365529e54)
